samples/nrf9160/serial_lte_modem/src/main.c

```diff
@@ -0,0 +1,289 @@
+		/* Read AT socket in non-blocking mode. */
+		r_bytes = recv(at_socket_fd, at_read_buff,
+			sizeof(at_read_buff), MSG_DONTWAIT);
+		k_mutex_unlock(&socket_mutex);
+
+		/* Forward the data over UART if any. */
+		/* If no data, errno is set to EGAIN and we will try again. */
+		if (r_bytes > 0) {
+			/* Forward the data over UART if any. */
+			if (at_cmd_responded) {
+				(void)inter_connect_notify(NOT_TYPE_AT,
+					at_read_buff,
+					r_bytes);
+			} else {
+				(void)inter_connect_send(
+					RSP_TYPE_BASE|CMD_TYPE_AT_CMD,
+					at_read_buff,
+					r_bytes);
+				at_cmd_responded = true;
+			}
+		}
+	}
+}
+
+/**@brief UART RX data handling */
+static void data_handler(u8_t cmd_type, const u8_t *data_buf, u8_t data_len)
+{
+	if (cmd_type == CMD_TYPE_DATA) {
+		LOG_INF("%s", log_strdup(data_buf));
+		/* Echo back data for UART testing. */
+		(void)inter_connect_send(CMD_TYPE_DATA, "nRF91 ready", 11);
+	} else if (cmd_type == CMD_TYPE_AT_CMD) {
+		if (data_len <= CONFIG_THIN_MODEM_UART_BUF_SIZE) {
+			memset(at_buf, 0x00, CONFIG_THIN_MODEM_UART_BUF_SIZE);
+			memcpy(at_buf, data_buf, data_len);
+			at_buf_len = data_len;
+			k_work_submit(&at_cmd_send_work);
+			at_cmd_responded = false;
+		} else {
+			LOG_WRN("Invalid data dropped");
+		}
+	}
+#if defined(CONFIG_THIN_GPIO_WAKEUP)
+	else if (cmd_type == (RSP_TYPE_BASE|CMD_TYPE_SYNC_CMD)) {
+		LOG_INF("Sync up");
+	}
+#endif
+#if defined(CONFIG_THIN_GPS_SERVICE)
+	else if (cmd_type == CMD_TYPE_GPS_CMD) {
+		th_gps_control(cmd_type, data_buf);
+	}
+#endif
+#if defined(CONFIG_THIN_SNTP_SERVICE)
+	else if (cmd_type == CMD_TYPE_TIME_CMD) {
+		th_sntp_control(cmd_type);
+	}
+#endif
+#if defined(CONFIG_LTE_LINK_CONTROL)
+	else if ((cmd_type&CMD_TYPE_MDM_BASE) == CMD_TYPE_MDM_BASE) {
+		th_modem_control(cmd_type, data_buf);
+	}
+#endif
+#if defined(CONFIG_THIN_MQTT_SERVICE)
+	else if ((cmd_type&CMD_TYPE_MQTT_BASE) == CMD_TYPE_MQTT_BASE) {
+		th_mqtt_control(cmd_type, data_buf, data_len);
+	}
+#endif
+#if defined(CONFIG_THIN_LWM2M_SERVICE)
+	else if ((cmd_type&CMD_TYPE_LWM2M_BASE) == CMD_TYPE_LWM2M_BASE) {
+		th_lwm2m_control(cmd_type, data_buf, data_len);
+	}
+#endif
+	else {
+		LOG_WRN("unknown data, dropped");
+	}
+}
+
+/**@brief Recoverable BSD library error. */
+void bsd_recoverable_error_handler(uint32_t err)
+{
+	LOG_ERR("bsdlib recoverable error: %u", err);
+}
+
+/**@brief Irrecoverable BSD library error. */
+void bsd_irrecoverable_error_handler(uint32_t err)
+{
+	LOG_ERR("bsdlib irrecoverable error: %u", err);
+	__ASSERT_NO_MSG(false);
+}
+
+void start_execute(void)
+{
+	int err;
+
+#if !defined(CONFIG_THIN_TEST)
+	/* Initialize the inter connect library. */
+	err = inter_connect_init(data_handler);
+	if (err) {
+		LOG_ERR("Init inter_connect error: %d", err);
+		return;
+	}
+#endif /* CONFIG_THIN_TEST*/
+
+	/* AT socket creation. Handle must be valid to continue. */
+	at_socket_fd = socket(AF_LTE, 0, NPROTO_AT);
+	fds[0].fd = at_socket_fd;
+	fds[0].events = ZSOCK_POLLIN;
```

Comment:
I'd stick to POSIX naming here - `POLLIN`. And it'd make sense to use the socket description (in fds) after it's checked it's valid (just below).

Reply:
Updated, code had referred to at_host in NCS v0.4.0.